Fabric.Terminology.SqlServer/Persistence/SqlClientTermValueSetRepository.UoWOperation.cs

```diff
@@ -0,0 +1,138 @@
+﻿namespace Fabric.Terminology.SqlServer.Persistence
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using CallMeMaybe;
+
+    using Fabric.Terminology.Domain;
+    using Fabric.Terminology.Domain.Models;
+    using Fabric.Terminology.Domain.Persistence;
+    using Fabric.Terminology.SqlServer.Models.Dto;
+    using Fabric.Terminology.SqlServer.Persistence.UnitOfWork;
+
+    internal partial class SqlClientTermValueSetRepository
+    {
+        private static ValueSetCodeDto ConvertToValueSetCodeDto(Guid valueSetGuid, ICodeSystemCode code) =>
+            new ValueSetCodeDto(code) { ValueSetGUID = valueSetGuid };
+
+        private AddRemoveCodeOperations PrepareAddRemoveCodes(
+            Guid valueSetGuid,
+            IReadOnlyCollection<ICodeSystemCode> codesToAdd,
+            IReadOnlyCollection<ICodeSystemCode> codesToRemove)
+        {
+            var currentCodeDtos = this.uowManager.GetCodeDtos(valueSetGuid);
+
+            //// There could be duplicates between the codesToAdd and codesToRemove collections.
+            //// In cases where a code is found in both lists, it will simply be removed from each list
+            //// since we are using SqlBulkCopy to batch insert records the "added" codes (e.g. there is no
+            //// update/delete capability in that process and we want to ensure we do not insert duplicate codes)
+
+            // dups are codes that would be removed and then immediately added again.
+            var dups = codesToAdd.Where(code => codesToRemove.Select(r => r.CodeGuid).Contains(code.CodeGuid)).ToList();
```

Comment:
This is an `O(n²)` operation. You could either put `codesToRemove.Select(r => r.CodeGuid)` into a HashSet ahead of time, or use a LINQ Join operation to reduce the complexity.